src/main/java/com/autotune/Autotune.java

```diff
@@ -154,6 +161,29 @@ public static void main(String[] args) {
 
     }
 
+    /**
+     * Set up the data sources available at installation time from config file
+     */
+    private static void setUpDataSources() {
+        DataSourceCollection dataSourceCollection = DataSourceCollection.getInstance();
+        dataSourceCollection.addDataSourcesFromConfigFile(KruizeConstants.CONFIG_FILE);
```

Comment:
Don't we have to read from the DB for all the existing the datasources at startup time? Remember that datasources can be added by the user after startup as well through the API

Reply:
@dinogun It's for the first time setup while adding datasources from YAML file. 

For fetching the data sources we can call `list_datasource` API added by shreya that will load all datasources from db.

Reply:
I think we can call `list_datasources` api while instantiating the dataSourceCollection object to make it sync with db.  Should I add it as a TODO item if it is okay?

Reply:
Added `loadDataSourcesFromDB()` to fetch available datasources from DB. 

**Kruize Pod Logs (On Initialization):** 
```java
2024-04-1711:50:51.666 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1711:50:52.665 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !
2024-04-1711:50:52.792 INFO [main][Autotune.java(251)]-DB creation successful !
2024-04-1711:50:52.793 INFO [main][Autotune.java(259)]-DB Liveliness probe connection successful!
2024-04-1711:50:52.793 INFO [main][Autotune.java(109)]-Now running kruize local DDL's 
2024-04-1711:50:52.845 INFO [main][Autotune.java(251)]-DB creation successful !
2024-04-1711:50:52.846 INFO [main][Autotune.java(259)]-DB Liveliness probe connection successful!
2024-04-1711:50:52.847 INFO [main][DataSourceCollection.java(47)]-Checking available datasources from database:
2024-04-1711:50:53.005 INFO [main][DataSourceCollection.java(50)]-No datasource found in database.
2024-04-1711:50:53.044 INFO [main][DataSourceCollection.java(89)]-Trying to add the datasource to collection: prometheus-1
2024-04-1711:50:53.045 INFO [main][DataSourceCollection.java(97)]-Verifying datasource reachability status: prometheus-1
2024-04-1711:50:53.201 INFO [main][DataSourceCollection.java(100)]-Datasource is serviceable.
2024-04-1711:50:53.234 INFO [main][DataSourceCollection.java(104)]-Datasource added to the DB successfully.
2024-04-1711:50:53.234 INFO [main][DataSourceCollection.java(109)]-Datasource added to the collection successfully.
2024-04-1711:50:53.234 INFO [main][Autotune.java(188)]-Checking available datasources:
2024-04-1711:50:53.234 INFO [main][Autotune.java(194)]-Datasource found: prometheus-1, http://prometheus-k8s.monitoring.svc.cluster.local:9090
2024-04-1711:50:53.238 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1711:50:53.338 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !
```


**Kruize Pod Logs (On Restart):** 
```java
2024-04-1711:51:23.966 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1711:51:24.864 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !
2024-04-1711:51:24.931 WARN [main][Autotune.java(240)]-sql: alter table if exists kruize_experiments add constraint UK_experiment_name unique (experiment_name); failed due to : Converting `org.hibernate.exception.SQLGrammarException` to JPA `PersistenceException` : JDBC exception executing SQL [alter table if exists kruize_experiments add constraint UK_experiment_name unique (experiment_name);]
2024-04-1711:51:24.936 INFO [main][Autotune.java(251)]-DB creation successful !
2024-04-1711:51:24.937 INFO [main][Autotune.java(259)]-DB Liveliness probe connection successful!
2024-04-1711:51:24.937 INFO [main][Autotune.java(109)]-Now running kruize local DDL's 
2024-04-1711:51:24.940 WARN [main][Autotune.java(240)]-sql: alter table kruize_experiments add column metadata_id bigint references kruize_dsmetadata(id), alter column datasource type varchar(255); failed due to : Converting `org.hibernate.exception.SQLGrammarException` to JPA `PersistenceException` : JDBC exception executing SQL [alter table kruize_experiments add column metadata_id bigint references kruize_dsmetadata(id), alter column datasource type varchar(255);]
2024-04-1711:51:24.941 INFO [main][Autotune.java(251)]-DB creation successful !
2024-04-1711:51:24.941 INFO [main][Autotune.java(259)]-DB Liveliness probe connection successful!
2024-04-1711:51:24.942 INFO [main][DataSourceCollection.java(47)]-Checking available datasources from database:
2024-04-1711:51:25.136 INFO [main][DataSourceCollection.java(53)]-Datasource found: prometheus-1
2024-04-1711:51:25.179 ERROR [main][DataSourceCollection.java(146)]-Datasource: prometheus-1 already exists!
2024-04-1711:51:25.180 INFO [main][Autotune.java(188)]-Checking available datasources:
2024-04-1711:51:25.180 INFO [main][Autotune.java(194)]-Datasource found: prometheus-1, http://prometheus-k8s.monitoring.svc.cluster.local:9090
2024-04-1711:51:25.184 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1711:51:25.273 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !
```

Reply:
**Update Logs After Restart (With Updated Error Msgs):** 
```java
2024-04-1712:39:53.305 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1712:39:54.239 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !
2024-04-1712:39:54.302 WARN [main][Autotune.java(240)]-sql: alter table if exists kruize_experiments add constraint UK_experiment_name unique (experiment_name); failed due to : add constraint - Attempted operation has already been executed
2024-04-1712:39:54.308 INFO [main][Autotune.java(253)]-DB creation successful !
2024-04-1712:39:54.309 INFO [main][Autotune.java(261)]-DB Liveliness probe connection successful!
2024-04-1712:39:54.309 INFO [main][Autotune.java(109)]-Now running kruize local DDL's 
2024-04-1712:39:54.312 WARN [main][Autotune.java(242)]-sql: alter table kruize_experiments add column metadata_id bigint references kruize_dsmetadata(id), alter column datasource type varchar(255); failed due to : add column - Attempted operation has already been executed
2024-04-1712:39:54.313 INFO [main][Autotune.java(253)]-DB creation successful !
2024-04-1712:39:54.313 INFO [main][Autotune.java(261)]-DB Liveliness probe connection successful!
2024-04-1712:39:54.315 INFO [main][DataSourceCollection.java(47)]-Checking available datasources from database:
2024-04-1712:39:54.500 INFO [main][DataSourceCollection.java(53)]-Datasource found: prometheus-1
2024-04-1712:39:54.541 ERROR [main][DataSourceCollection.java(146)]-Datasource: prometheus-1 already exists!
2024-04-1712:39:54.542 INFO [main][Autotune.java(188)]-Checking available datasources:
2024-04-1712:39:54.542 INFO [main][Autotune.java(194)]-Datasource found: prometheus-1, http://prometheus-k8s.monitoring.svc.cluster.local:9090
2024-04-1712:39:54.545 INFO [main][KruizeHibernateUtil.java(63)]-DB is trying to connect to jdbc:postgresql://postgres-service:5432/kruizeDB
2024-04-1712:39:54.630 INFO [main][KruizeHibernateUtil.java(65)]-DB build session is successful !

```